src/OnboardingSPA/components/Drawer/DrawerPanel/DesignHeaderMenu.js

```diff
@@ -1,17 +1,139 @@
+	const [ headerMenuPreviewData, setHeaderMenuPreviewData ] = useState();
+	const [ selectedPattern, setSelectedPattern ] = useState( '' );
+
+	const { currentStep, currentData, themeStatus } =
+		useSelect( ( select ) => {
+			return {
+				currentStep: select( nfdOnboardingStore ).getCurrentStep(),
+				currentData:
+					select( nfdOnboardingStore ).getCurrentOnboardingData(),
+				themeStatus: select( nfdOnboardingStore ).getThemeStatus(),
+			};
+		}, [] );
+
+	const {
+		setCurrentOnboardingData,
+		updateThemeStatus,
+		setHeaderMenuData,
+	} = useDispatch( nfdOnboardingStore );
+
+	const getPatternsData = async () => {
+		const headerMenuPreviewResponse = await getPatterns( currentStep.patternId );
+		if ( headerMenuPreviewResponse?.error ) {
+			return updateThemeStatus( THEME_STATUS_NOT_ACTIVE );
+		}
+		setHeaderMenuPreviewData(headerMenuPreviewResponse.body);
+
+		// let headerMenuPatterns = [];
+		// headerMenuPreviewResponse.body.forEach( ( pageParts ) => {
+		// 	if(headerMenuSlugs.includes(pageParts.slug)){
+		// 		console.log(pageParts.content);
+		// 		headerMenuPatterns.push(pageParts.content);
+		// 	}
+		// });
+		// console.log(headerMenuPatterns);
+
+		const patternsResponse = await getHeaderMenuPatterns();
+		if ( patternsResponse?.error ) {
+			return updateThemeStatus( THEME_STATUS_NOT_ACTIVE );
+		}
+        setPatterns( patternsResponse?.body );
+
+		if(!currentData.data.partHeader || currentData.data.partHeader == "") {
+			const defaultHeaderMenu = await getDefaultHeaderMenu();
+			if ( defaultHeaderMenu?.error ) {
+				return updateThemeStatus( THEME_STATUS_NOT_ACTIVE );
+			}
+			currentData.data.partHeader = defaultHeaderMenu.body;
+			setCurrentOnboardingData( currentData );
+		}
+		setSelectedPattern(currentData.data.partHeader);
+
+		let [pageContent, headerContent, pagePreview] = ["", "", ""];
+		headerMenuPreviewResponse.body.forEach( ( pageParts ) => {
+			if(headerMenuBodySlugs.includes(pageParts.slug)){
+				pageContent += pageParts.content;
+			}
+			if(pageParts.slug === currentData.data.partHeader){
+				headerContent += pageParts.content;
+			}
+		});
+		pagePreview = headerContent + pageContent;
+		setHeaderMenuData( pagePreview );
+		setIsLoaded( true );
+	};
+
+	useEffect( () => {
+		if ( ! isLoaded && themeStatus === THEME_STATUS_ACTIVE )
+			getPatternsData();
+	}, [ isLoaded, themeStatus ] );
+
+	const handleClick = ( idx ) => {
+		const selectedPattern = patterns[ idx ];
+
+		setSelectedPattern( selectedPattern.slug );
+		currentData.data.partHeader = selectedPattern.slug;
+		setCurrentOnboardingData( currentData );
+
+		let newPagePattern = selectedPattern.content;
+		headerMenuPreviewData.forEach( ( pageParts ) => {
+			if(headerMenuBodySlugs.includes(pageParts.slug)){
+				newPagePattern += pageParts.content;
+			}
+		});
+		setHeaderMenuData( newPagePattern );
+	};
+
+	const buildPreviews = () => {
+		return patterns?.map( ( pattern, idx ) => {
+			return (
+				<HeaderMenuPreview
```

Comment:
There is a delay for me on the Header Menu page before anything shows up and that was fixed for other steps with LivePreviewSkeleton component not the default skeleton inside LivePreview, Maybe can we use that and inline the number of previews?